### ISSUE TITLE
[Bug]: Wallpaper isnt loading

### ISSUE THREAD

**Issue by Swordpro3001** [prasanthrangan/hyprdots#1165]
### Description

I reinstalled Arch so I had to reinstall hyprdots. After the reinstall and reboot the wallpaper didnt load. If I try to change the wallpaper with Super+alt+left it doesnt change.

### Steps to Reproduce

1. Install Arch
2. Update Hyprland-Header
3. Pull and install Hyprdots

If I try to change the wallpaper with `Hyprdots wallpaper set` its giving:
'Missing /Path/to/Wallpaper'

With `Hyprdots wallpaper next`:
`Error: "Failed to connect to socket: Connection refused (os error 111)"`

If I try to start swww with `swww-daemon`:
`Error: "couldn't bin socket: Address already in use (os error 98)"` even when swww isnt running in htop.
  
### Expected Behavior

The wallpaper should be loading after reboot or after setting the wallpaper.

### Actual Behavior

The wallpaper is grey.

### Screenshot

![Screenshot](https://github.com/prasanthrangan/hyprdots/assets/107194701/7b3e74e2-8f46-44d8-9f14-bee95cd4391e)

### Environment

- Hardware: Laptop: Acer Nitro 517-41
- Linux Distro: ArchLinux
- Linux Version, archLinux 6.8.1-arch1-1
- Hyprland Version, v0.37.1-47
- Installation Environment: Dualboot Laptop
- Other info: Resolution: 1920x1080p 60hz

**Comment by kRHYME7** [comment]
try to run ``` swww init ```

**Comment by Swordpro3001** [comment]
> try to run `swww init`

It says:
`DEPRECATION WARNING: 'swww init' IS DEPRECATED, call 'swww-daemon' directly instead 
WARNING: socket file /run/user/1000/swww.socket was not deleted when the previos demon exited`

**Comment by Swordpro3001** [comment]
> > try to run `swww init`
> 
> It says: `DEPRECATION WARNING: 'swww init' IS DEPRECATED, call 'swww-daemon' directly instead WARNING: socket file /run/user/1000/swww.socket was not deleted when the previos demon exited`

Ok now its kinda weird.
If i run `swww init` it show the error above but it loads the wallpaper. **BUT** when I change theme it switches theme but the background is grey.

**Comment by kRHYME7** [comment]
Output for ``` yay -Qs swww ```  ?

**Comment by Swordpro3001** [comment]
> Output for `yay -Qs swww` ?

` local/swww-git 0.4.0.r317.g06ede52-1
    Efficient animated wallpaper daemon for Wayland, controlled at runtime
local/swww-git-debug 0.4.0.r317.g06ede52-1
    Detached debugging symbols for swww-git
local/waypaper-git 2.1.r13.gbf75172-1
    GUI wallpaper setter for both Wayland and X11 window managers that works as a frontend for popular backends like swaybg, swww, and feh.`

**Comment by kRHYME7** [comment]
You're on the latest commit, therefore it should work 

Output for ```  grep "swww-" ~/.config/hyprdots/scripts/swwwallpaper.sh ``` ?

**Comment by Swordpro3001** [comment]
> You're on the latest commit, therefore it should work
> 
> Output for ` grep "swww-" ~/.config/hyprdots/scripts/swwwallpaper.sh` ?

`swww-daemon --format xrgb`

**Comment by kRHYME7** [comment]
Alright try to add "&" on that line. 

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/a3a69feb-b917-43b7-86e4-224f71436f47)

**Comment by Swordpro3001** [comment]
> The latest commit should have fix it

Ok, It fixed that that if I boot it shows the wallpaper but it didn't fix the issue with changing the wallpaper too fast or change theme swww "crashes".

**Comment by kRHYME7** [comment]
> but it didn't fix the issue with changing the wallpaper too fast or change theme swww "crashes".

Can you please  elaborate more?

I tried the latest commit and it works on my end.  can you provide again the outputs you provide above https://github.com/prasanthrangan/hyprdots/issues/1165#issue-2203916433 ?

**Comment by Swordpro3001** [comment]
> Can you please elaborate more?

If i try to change the wallpaper to quickly with `Super+alt+left/right` the animation crashes and the wall paper becomes grey. Same with changing the theme but it crashes on every theme-change

> Hyprdots wallpaper next output for this ?

So if swww crashed `Hyprdots wallpaper next` is giving: `Error: "Failed to connect to socket: Connection refused (os error 111)"
Error: "couldn't bind socket: Address already in use (os error 98)"
Error: "Failed to connect to socket: Connection refused (os error 111)"`.

**Comment by prasanthrangan** [comment]
Hi, try clearing the cache directory `~/.cache/swww/`

**Comment by Swordpro3001** [comment]
> Hi, try clearing the cache directory `~/.cache/swww/`

removed all files from that directory and there are still the same problems

**Comment by andy-uplabs** [comment]
> > Hi, try clearing the cache directory `~/.cache/swww/`
> 
> removed all files from that directory, rebooted and there are still the same problems

try running both of these and rebooting

`rm -rf ~/.config/swww/*`

`rm -rf $XDG_RUNTIME_DIR/swww.socket`

**Comment by andres-gr** [comment]
> > Hi, try clearing the cache directory `~/.cache/swww/`
> 
> removed all files from that directory, rebooted and there are still the same problems

try running both of these and rebooting

rm -rf ~/.cache/swww/*

rm -rf $XDG_RUNTIME_DIR/swww.socket

**Comment by opticblu** [comment]
100% a bug, whenever you change a wallpaper swww-daemon dumps core. This only started after the most recent updates, the wallbashv4 changes perhaps? It may not be with hyprdots itself, but this only presented after hyprdots update, nothing else changed. After I noticed this regression, updated my system (usually current, only had a few things out of date) so:


```
swww-daemon --version
swww-daemon 0.9.1
```

```
Mar 23 21:05:54 systemd-coredump[16124]: [🡕] Process 3291 (swww-daemon) of user 1000 dumped core.
```


This used to happen when hyprdots was new, same exact thing: best guess is some cache issue.


Here's the error log after rm -rf ~/.cache/swww/* rm -rf $XDG_RUNTIME_DIR/swww.socket, then launching swww-daemon manually. It launches, displays wallpaper, but when you change wallpaper:


```
 02:13:00 [INFO] (transition) BumpPool with: 2 buffers. Size: 86400Kb
02:13:00 [INFO] (transition) BumpPool with: 2 buffers. Size: 86400Kb

thread 'rayon thread 2' has overflowed its stack
fatal runtime error: stack overflow

[1]  + 20286 IOT instruction (core dumped)  swww-daemon
```

The stack trace stuff in the system log is unremarkable

will try to find out more, could be hyprland-git branch related, gonna try to move to stable and see if it chokes

**Comment by kRHYME7** [comment]
@opticblu  Did the swww crash upon boot?

can you try to run ``` ~/.config/hyprdots/scripts/swwwallpaper.sh ```  instead of manually running ```   swww-daemon --format xrgb & ``` this will emulate this part 
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/499c7129-610b-42ca-9c1c-1f89f7d88079)
Just to see what the script do after boot up .

**Comment by opticblu** [comment]
think there's more to this story, seeing if swww-git fixes

https://github.com/LGFae/swww/commit/4288739a6f87b49518001abe1764fcfdf97004a8


```
    rayon::ThreadPoolBuilder::default()
        .thread_name(|i| format!("rayon thread {i}"))
        .stack_size(1 << 18) // 256KiB; we do not need a large stack
        .stack_size(1 << 19) // 512KiB; we do not need a large stack
        .build_global()
        .expect("failed to configure rayon global thread pool");
    make_logger(cli.quiet);
```


will update in a min

**Comment by opticblu** [comment]
> @opticblu Did the swww crash upon boot?
> 
> can you try to run `~/.config/hyprdots/scripts/swwwallpaper.sh` instead of manually running ` swww-daemon --format xrgb &` this will emulate this part ![image](https://private-user-images.githubusercontent.com/53417443/316279929-499c7129-610b-42ca-9c1c-1f89f7d88079.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyNDk3NjMsIm5iZiI6MTcxMTI0OTQ2MywicGF0aCI6Ii81MzQxNzQ0My8zMTYyNzk5MjktNDk5YzcxMjktNjEwYi00MmNhLTljMWMtMWY4OWY3ZDg4MDc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI0VDAzMDQyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE4ZTAxZTBkOTEwYjdjYTViODJjMDY3M2M1YWEwOTQ4ZDE1MjRhMGY1MjUwZjFiOTdiNzVlMDNkNDMxNDNiZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gGlZpP2IYHE4m3QTz7VVXCbCm2cNbfy-eAsDRZHPzZw) Just to see what the script do after boot up .

my reply above was correct, moving to AUR/swww-git fixed this,  I did rm -rf ~/.cache/* (low key w/ sudo; in zfs i trust) after I moved to swww-git, then reinstalled hyprdots, just config (in scripts dir: ./install.sh -r), and rebooted

upon reboot, appears to work, with and without wallbash enabled

TL;DR use swww-git 
(lol to quote the lgfae, the swww dev: "Shit, I should have know better than to aggressively reduce stack size like that."https://github.com/LGFae/swww/issues/239#issuecomment-2016533591) 
+also yeet your cache, probably a hidpi thing still

**Comment by prasanthrangan** [comment]
I'm not facing any issues with my system... tried both non-git version and git version 0.9.1

**Comment by opticblu** [comment]
> I'm not facing any issues with my system... tried both non-git version and git version 0.9.1

I think it's either a hidpi issue or a fractional scaling issue? something similar used to happen about a year? or so ago, to the point where i had an exec-once = fixswww.sh on startup (just a--hesitate to even call it a-- script with rm -rf ~/.cache/swww/* && rm -rf $XDG_RUNTIME_DIR/swww.socket or something to that effect, probably in my old zfs snapshots, can't remember.

it does seem to resolve with 

rm -rf ~/.cache/swww/*

rm -rf $XDG_RUNTIME_DIR/swww.socket

not a big deal for me, can exec-once a cache clear, just FYI

**Comment by kRHYME7** [comment]
how about ``` swww clear-cache ``` ?

**Comment by prasanthrangan** [comment]
> rm -rf ~/.cache/swww/* && rm -rf $XDG_RUNTIME_DIR/swww.socket

Glad its working for you... maybe we should think about clearing the swww cache at startup.

**Comment by opticblu** [comment]
it hasn't done it in forever, only restarted after that wallbash push today/yesterday

my underlying config is supe... err hypr-weird: incus gui containers w/ wayland socket, zfs root/kmod/clang kernel, apparmor, multi-gpu, vertical 4k monitor, etc

in my case, it could be anything, but if others are experiencing it, cache clear might not hurt

**Comment by prasanthrangan** [comment]
> it hasn't done it in forever, only restarted after that wallbash push today/yesterday
> 
> my underlying config is supe... err hypr-weird: incus gui containers w/ wayland socket, zfs root/kmod/clang kernel, apparmor, multi-gpu, vertical 4k monitor, etc
> 
> the point being: in my case, it could be anything, but if others are experiencing it, cache clear might not hurt


woww you've got one of a kind setup :)
I just have a single ultrawide monitor so never had these issues with swww...
anyways lets wait and see if others have such issues!

**Comment by Swordpro3001** [comment]
> @opticblu Did the swww crash upon boot?
> 
> can you try to run `~/.config/hyprdots/scripts/swwwallpaper.sh` instead of manually running ` swww-daemon --format xrgb &` this will emulate this part ![image](https://private-user-images.githubusercontent.com/53417443/316279929-499c7129-610b-42ca-9c1c-1f89f7d88079.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEyNjE4OTUsIm5iZiI6MTcxMTI2MTU5NSwicGF0aCI6Ii81MzQxNzQ0My8zMTYyNzk5MjktNDk5YzcxMjktNjEwYi00MmNhLTljMWMtMWY4OWY3ZDg4MDc5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzI0VDA2MjYzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdjNTkxMGM3OWI5YmZkMzhjMWUxMmM5ZjkyMzUwMjNlYzVmNmUwZjQyMWY4MDViMjE0ZGEyNmYzYTUzYThiOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HX2J9IXL41b41GfhupfL41ewwoQbX6iaKkJQkh6p6qk) Just to see what the script do after boot up .

Tried that same Error

**Comment by Swordpro3001** [comment]
> > I'm not facing any issues with my system... tried both non-git version and git version 0.9.1
> 
> I think it's either a hidpi issue or a fractional scaling issue? something similar used to happen about a year? or so ago, to the point where i had an exec-once = fixswww.sh on startup (just a--hesitate to even call it a-- script with rm -rf ~/.cache/swww/* && rm -rf $XDG_RUNTIME_DIR/swww.socket or something to that effect, probably in my old zfs snapshots, can't remember.
> 
> it does seem to resolve with
> 
> rm -rf ~/.cache/swww/*
> 
> rm -rf $XDG_RUNTIME_DIR/swww.socket
> 
> not a big deal for me, can exec-once a cache clear, just FYI

Ok that kinda worked, but when I switch themes it still crashes.
But thanks

**Comment by Swordpro3001** [comment]
> I'm not facing any issues with my system... tried both non-git version and git version 0.9.1

Maybe because I have the git version? I installed it from git because curl wont install the version in the hyprdots-script.

**Comment by opticblu** [comment]
> > > I'm not facing any issues with my system... tried both non-git version and git version 0.9.1
> > 
> > 
> > I think it's either a hidpi issue or a fractional scaling issue? something similar used to happen about a year? or so ago, to the point where i had an exec-once = fixswww.sh on startup (just a--hesitate to even call it a-- script with rm -rf ~/.cache/swww/* && rm -rf $XDG_RUNTIME_DIR/swww.socket or something to that effect, probably in my old zfs snapshots, can't remember.
> > it does seem to resolve with
> > rm -rf ~/.cache/swww/*
> > rm -rf $XDG_RUNTIME_DIR/swww.socket
> > not a big deal for me, can exec-once a cache clear, just FYI
> 
> Ok that kinda worked, but when I switch themes it still crashes. But thanks

when you switch themes or switch wallpapers? 

looks like some config changes were just pushed to the create_cache script, just glanced but looks like that the imagemagick part (cpu/mem intensive part that might cause this) of the script has some shiny new &> /dev/null amendments, presume that's a mitigation for this, 100% a guess though, not too familiar

will try and update with latest change and report back

**Comment by Swordpro3001** [comment]
>I pulled the latest changes and ran the script again, then did "Hyprdots cache rebuild" (hyprdots-ctl-git, not sure if it's included in the script tbh), now theme/wall changes work fine, even after logout/login, give it a shot, think it will work 

After pulling the repo, running the install script, rebuilding the cache and rebooting it still crashes after switching themes. But thank you for trying it, it shouldnt be a problem cause I can "restart" swww with `swww init`.
 <3 <3

**Comment by kRHYME7** [comment]
@sandeng

**Comment by kRHYME7** [comment]
try to ``` swww clear cache ``` then change wallpapers

**Comment by kRHYME7** [comment]
I'll be moving this to discussions for now to utilize reply/threads 

Issue is fixed https://github.com/LGFae/swww/commit/820ea906acf0f9d86e808303917c1f18ae914d2c
